lib/services/access_checker.go

```diff
@@ -321,24 +322,35 @@ func (a *accessChecker) CheckAccess(r AccessCheckable, state AccessState, matche
 func (a *accessChecker) GetKubeResources(cluster types.KubeCluster) (allowed, denied []types.KubernetesResource) {
 	if len(a.info.AllowedResourceIDs) > 0 {
 		for _, r := range a.info.AllowedResourceIDs {
-			if r.Kind == types.KindKubePod && r.Name == cluster.GetName() && r.ClusterName == a.localCluster {
+			if r.Name != cluster.GetName() || r.ClusterName != a.localCluster {
+				continue
+			}
+			switch {
+			case slices.Contains(types.KubernetesResourcesKinds, r.Kind):
 				splitted := strings.SplitN(r.SubResourceName, "/", 3)
 				// This condition should never happen since SubResourceName is validated
 				// but it's better to validate it.
 				if len(splitted) != 2 {
 					continue
 				}
+				// When the user is granted access through a Pod resource access request,
+				// he has no denied resources, which results in the denied being an empty slice.
```

Comment:
it's not clear to me why he has no denied resources, shouldn't we always consider the full implications of the user's roleset and apply any resource access requests as a further restriction? A resource access request should never be able to bypass a deny rule

Reply:
We evaluated the deny rule when we created the resource access request. 

If you have resources of `kind=pod`:

- /<teleport-cluster>/pod/<kube-cluster>/<namespace>/<pod>

Given that you already have an access request to pod(s), you can only access those pod(s) which results in the deny set being empty because the allow set only includes the pods you requested access to, and at creation time your `search_as_role` allowed the request.

Reply:
If the requested pod is in a leaf cluster then I don't believe access to it will have been validated at all, you can request access to arbitrary resources in leaf clusters, we rely on the access check to be done in the leaf cluster at access time

it's also possible that any of the users role specs have changed since the request was created